server/reports/expiring-items/2_3_0/convert_data_js/convert_data.test.ts

```diff
@@ -0,0 +1,111 @@
+    expect(calculateDaysUntilExpired("2024-04-03")).toBe(2);
+    expect(calculateDaysUntilExpired("2024-05-03")).toBe(32);
+  });
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+});
+
+describe("calculate expected usage", () => {
+  it("returns undefined if either are undefined", () => {
+    expect(calculateExpectedUsage(undefined, undefined)).toBe(undefined);
+    expect(calculateExpectedUsage(undefined, 1)).toBe(undefined);
+    expect(calculateExpectedUsage(1, undefined)).toBe(undefined);
+  });
+
+  it("returns expected usage if both defined", () => {
+    expect(calculateExpectedUsage(20, 500)).toBe(333);
+  });
+});
+
+describe("calculate stock at risk ", () => {
+  it("returns undefined if packSize, totalNumberOfPacks, OR expiryDate is undefined", () => {
+    expect(calculateStockAtRisk(undefined, 1, 1, 1)).toBe(undefined);
+    expect(calculateStockAtRisk(1, undefined, 1, 1)).toBe(undefined);
+    expect(calculateStockAtRisk(1, 1, 1, undefined)).toBe(undefined);
+  });
+
+  it("returns undefined if averageMonthlyConsumption is undefined AND expiryDate is in the future", () => {
+    expect(calculateStockAtRisk(2, 100, undefined, 1)).toBe(undefined);
+  });
+
+  it("returns all stock if averageMonthlyConsumption is undefined AND expiryDate is in the past", () => {
+    expect(calculateStockAtRisk(2, 100, undefined, -1)).toBe(200);
+  });
+
+  it("returns stock at risk as all stock minus what we will consume before expiry date", () => {
+    expect(calculateStockAtRisk(2, 100, 10, 60)).toBe(180);
+  });
+});
+
+describe("test round days to integer", () => {
+  it("returns undefined if undefined", () => {
+    expect(roundDaysToInteger(undefined)).toBe(undefined);
+  });
+
+  it("returns rounded value if defined", () => {
+    expect(roundDaysToInteger(2.1)).toBe(2);
+    expect(roundDaysToInteger(2.11)).toBe(2);
+    expect(roundDaysToInteger(0.123)).toBe(0);
+    expect(roundDaysToInteger(2)).toBe(2);
+  });
+});
+
+// describe("test mocked functions", () => {
+//   it("calls roundDaysToInteger with the correct argument", () => {
+//     const spyProcessStockLines = jest.spyOn(
+//       require("./convert_data"),
+//       "processStockLines"
+//     );
+//     spyProcessStockLines.mockImplementation((x) => x);
+//     const spyCalculateDaysUntilExpired = jest.spyOn(
+//       require("./convert_data"),
+//       "calculateDaysUntilExpired"
+//     );
+//     spyCalculateDaysUntilExpired.mockImplementation((x) => x);
+//     let res = processStockLines(inputData.stockLines.nodes);
+//     expect(spyCalculateDaysUntilExpired).toHaveBeenCalledTimes(2);
+//     expect(spyProcessStockLines).toHaveBeenCalledTimes(1);
+//     // expect(mockRoundDaysToInteger).toHaveBeenCalledWith(2.3);
+//     // expect(mockRoundDaysToInteger).toHaveBeenCalledWith(3.1);
+//     // expect(mockRoundDaysToInteger).toHaveBeenCalledWith(5.5);
+//     spyCalculateDaysUntilExpired.mockRestore();
+//   });
+// });
```

Comment:
I really wanted to get this mocked function testing working to ensure functions were being called correctly, but I couldn't figure this out.

Left here as comments in case reviewer is more familiar with this part of Jest!